### PR TITLE
Test Python 3.11-dev

### PR DESCRIPTION
**Pull request by hugovk**
Changes proposed in this pull request:

 * The first Python 3.11 beta is out next week: 2022-05-06
 * The full release is due out in October: 2022-10-03
 * Let's start testing the dev version
 * Like with `3.10-dev` last year, NumPy doesn't yet build for macOS/Ubuntu
 * https://peps.python.org/pep-0664/#schedule

CI targets table:

* `macos-latest` currently uses macOS Big Sur 11
* `windows-latest` currently uses Windows Server 2022
* https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources

### CODE REVIEW

File: docs/installation.rst [hunk 1]
```diff
@@ -465,11 +465,13 @@ These platforms are built and tested for every change.
...
 | Gentoo                           | 3.9                        | x86-64              |
 +----------------------------------+----------------------------+---------------------+
-| macOS 10.15 Catalina             | 3.7, 3.8, 3.9, 3.10, PyPy3 | x86-64              |
+| macOS 11 Big Sur                 | 3.7, 3.8, 3.9, 3.10, 3.11  | x86-64              |
+|                                  | PyPy3                      |                     |
 +----------------------------------+----------------------------+---------------------+
 | Ubuntu Linux 18.04 LTS (Bionic)  | 3.9                        | x86-64              |
 +----------------------------------+----------------------------+---------------------+
-| Ubuntu Linux 20.04 LTS (Focal)   | 3.7, 3.8, 3.9, 3.10, PyPy3 | x86-64              |
+| Ubuntu Linux 20.04 LTS (Focal)   | 3.7, 3.8, 3.9, 3.10, 3.11  | x86-64              |
```
Comment (radarhere) [inline docs/installation.rst:473]:
```suggestion
| Ubuntu Linux 20.04 LTS (Focal)   | 3.7, 3.8, 3.9, 3.10, 3.11, | x86-64              |
```
The two lines are joined into one when the RST is rendered - https://github.com/python-pillow/Pillow/blob/b4dab3aca301f49dd937fd29a06f1766cf138493/docs/installation.rst

File: docs/installation.rst [hunk 2]
```diff
@@ -478,7 +480,8 @@ These platforms are built and tested for every change.
 +----------------------------------+----------------------------+---------------------+
 | Windows Server 2016              | 3.7                        | x86-64              |
 +----------------------------------+----------------------------+---------------------+
-| Windows Server 2019              | 3.7, 3.8, 3.9, 3.10, PyPy3 | x86, x86-64         |
+| Windows Server 2022              | 3.7, 3.8, 3.9, 3.10, 3.11  | x86, x86-64         |
```
Comment (radarhere) [inline docs/installation.rst:483]:
```suggestion
| Windows Server 2022              | 3.7, 3.8, 3.9, 3.10, 3.11, | x86, x86-64         |
```

File: docs/installation.rst [hunk 3]
```diff
@@ -465,11 +465,13 @@ These platforms are built and tested for every change.
 +----------------------------------+----------------------------+---------------------+
 | Gentoo                           | 3.9                        | x86-64              |
 +----------------------------------+----------------------------+---------------------+
-| macOS 10.15 Catalina             | 3.7, 3.8, 3.9, 3.10, PyPy3 | x86-64              |
+| macOS 11 Big Sur                 | 3.7, 3.8, 3.9, 3.10, 3.11  | x86-64              |
```
Comment (radarhere) [inline docs/installation.rst:468]:
```suggestion
| macOS 11 Big Sur                 | 3.7, 3.8, 3.9, 3.10, 3.11, | x86-64              |
```

**Review by radarhere — APPROVED** [review]

**Review by hugovk — COMMENTED** [review]
Fix for RST rendering

**Review by radarhere — APPROVED** [review]

**Pull request merged**